components/selection/selection-mixin.js

```diff
@@ -87,6 +91,19 @@ export const SelectionMixin = superclass => class extends RtlMixin(superclass) {
 		return new SelectionInfo(keys, state);
 	}
 
+	setSelectionForAll(selected) {
```

Comment:
Exposing this for:
- the `selection-select-all` component, since it can't fire an event if it's external to the collection
- For filter's "clear" button to use
  - From a convo in #wc-list, Jeff doesn't think other use cases will need a clear button (rather than just uncheck the select all box).  So the mixin will still control how to clear, but there won't be a specific `selection-action-clear` control (at least not yet).

Reply:
If a consumer wants to set the selection - how would they go about it?  I think my original thinking was that they would set the `selected` property on the items individually.  I am just contemplating whether we would even want a `setSelectionInfo` (akin to the existing `getSelectionInfo`), and if so, would that method be used for this purpose instead.  Just thinking out loud.

Reply:
> I think my original thinking was that they would set the selected property on the items individually.

That was my thought as well, since they should be aware of their collection and can set the `input-selection` or `list-item` value directly.  Whereas the selection components don't know much about the list or how to traverse it or update it.  We could offer a `setSelectionInfo` to make that cleaner, but I still think `select-all` would need a `setSelectionInfoForAll` function (that could call `setSelectionInfo`), since it doesn't know all the keys?